hedera-node/src/main/java/com/hedera/services/ledger/AccountsCommitInterceptor.java

```diff
@@ -0,0 +1,59 @@
+package com.hedera.services.ledger;
+
+import com.hedera.services.context.SideEffectsTracker;
+import com.hedera.services.ledger.properties.AccountProperty;
+import com.hedera.services.state.merkle.MerkleAccount;
+import com.hederahashgraph.api.proto.java.AccountID;
+
+import java.util.List;
+import java.util.Map;
+
+public class AccountsCommitInterceptor implements
+		CommitInterceptor<AccountID, MerkleAccount, AccountProperty> {
+
+	// The tracker this interceptor should use for previewing changes. The interceptor is NOT
+	// responsible for calling reset() on the tracker, as that will be done by the client code.
+	private SideEffectsTracker sideEffectsTracker;
+
+	public AccountsCommitInterceptor(final SideEffectsTracker sideEffectsTracker) {
+		this.sideEffectsTracker = sideEffectsTracker;
+	}
+
+	/**
+	 * Accepts a pending change set, including creations and removals.
+	 *
+	 * @throws IllegalStateException if these changes are invalid
+	 */
+	@Override
+	public void preview(final List<MerkleLeafChanges<AccountID, MerkleAccount, AccountProperty>> changesToCommit) {
+		for (final var changeToCommit : changesToCommit) {
+			final var account = changeToCommit.id();
+			final var merkleAccount = changeToCommit.merkleLeaf();
+			final var changedProperties = changeToCommit.changes();
+
+			trackHBarTransfer(changedProperties, account, merkleAccount);
+		}
+
+		if (changesToCommit.size() > 1) {
```

Comment:
We still want to assert `sideEffectsTracker.getNetHbarChange() == 0` even if there is just one change. (What if the one change randomly increases an ℏ balance with no corresponding decrease? 😨)